packages/runtime/src/interpreter/opcode-list.ts

```diff
@@ -2259,3 +2260,123 @@ export class Swap extends Op {
     stack.push(b);
   }
 }
+
+// push the Nth value (0 indexed) from the top of the stack.
+// pops from stack: [...stack]
+// pushes to stack: [...stack, any (nth slot from top of stack)]
+// NOTE: dig 0 is same as dup
+export class Dig extends Op {
+  readonly line: number;
+  readonly depth: number;
+
+  /**
+   * Asserts 0 arguments are passed.
+   * @param args Expected arguments: [ depth ] // slot to duplicate
+   * @param line line number in TEAL file
+   */
+  constructor (args: string[], line: number) {
+    super();
+    this.line = line;
+    assertLen(args.length, 1, line);
+    assertOnlyDigits(args[0], line);
+
+    this.assertUint8(BigInt(args[0]), line);
+    this.depth = Number(args[0]);
+  };
+
+  execute (stack: TEALStack): void {
+    this.assertMinStackLen(stack, this.depth + 1, this.line);
+    const tempStack = new Stack<StackElem>(this.depth + 1); // depth = 2 means 3rd slot from top of stack
+    let copy;
+    for (let i = 0; i <= this.depth; i++) {
+      const top = stack.pop();
+      tempStack.push(top);
+      if (i === this.depth) { copy = top; }
+    }
+    while (tempStack.length()) { stack.push(tempStack.pop()); }
+    stack.push(copy as StackElem);
+  }
+}
+
+// selects one of two values based on top-of-stack: A, B, C -> (if C != 0 then B else A)
+// pops from stack: [...stack, {any A}, {any B}, {uint64 C}]
+// pushes to stack: [...stack, any (A or B)]
+export class Select extends Op {
+  readonly line: number;
+
+  /**
+   * Asserts 0 arguments are passed.
+   * @param args Expected arguments: [] // none
+   * @param line line number in TEAL file
+   */
+  constructor (args: string[], line: number) {
+    super();
+    this.line = line;
+    assertLen(args.length, 0, line);
+  };
+
+  execute (stack: TEALStack): void {
+    this.assertMinStackLen(stack, 3, this.line);
+    const toCheck = this.assertBigInt(stack.pop(), this.line);
+    const notZeroSelection = stack.pop();
+    const isZeroSelection = stack.pop();
+
+    if (toCheck !== 0n) { stack.push(notZeroSelection); } else { stack.push(isZeroSelection); }
+  }
+}
+
+/**
+ * push field F of the Ath transaction (A = top of stack) in the current group
+ * pops from stack: [...stack, uint64]
+ * pushes to stack: [...stack, transaction field]
+ * NOTE: "gtxns field" is equivalent to "gtxn _i_ field" (where _i_ is fetched from stack).
+ * gtxns exists so that i can be calculated, often based on the index of the current transaction.
+ */
+export class Gtxns extends Gtxn {
+  /**
+   * Sets `field`, `txIdx` values according to arguments passed.
+   * @param args Expected arguments: [transaction field]
+   * // Note: Transaction field is expected as string instead of number.
+   * For ex: `Fee` is expected and `0` is not expected.
+   * @param line line number in TEAL file
+   * @param interpreter interpreter object
+   */
+  constructor (args: string[], line: number, interpreter: Interpreter) {
+    super(["100", ...args], line, interpreter);
```

Comment:
why are we passing `100` here ?

Reply:
It's a mock value. didn't want to repeat code & reuse `Gtxn`. will add a comment